pkg/operatorsource/catalogsourceconfigcrud.go

```diff
@@ -0,0 +1,72 @@
+package operatorsource
+
+import (
+	"context"
+
+	v1 "github.com/operator-framework/operator-marketplace/pkg/apis/operators/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CatalogSourceConfigCRUD defines CRUD operations for CatalogSourceConfig object in the cluster
+type CatalogSourceConfigCRUD interface {
```

Comment:
Can we think of a better name than this? It should be a noun of some kind that describes the purpose of this type, not just the action(s) that it does. `ConfiguringHelper` maybe? @tkashem @aravindhp 

This comment goes for the type, the struct, and the file name.

Reply:
I agree with @kevinrizza. Moreover this shouldn't be in the `operatorsource` pkg. I also wonder if using `client-gen` to generate a client will help. Thoughts?

Reply:
I was a little uncomfortable with the name myself. If we have a unanimous decision to go with `ConfiguringHelper` I'm going with that. 
@aravindhp I'll need more info on `client-gen`

Reply:
Going with `client-gen` which seems to be the unanimous decision. Please do some research on that. There is also an old commit where @tkashem added that to this repo and was removed when we bumped group/version.